### DISCUSSION TITLE
Next release

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1822] [General]
If https://github.com/rust-bitcoin/rust-bitcoin/pull/1820 merges I think we should consider doing a release sooner than 6 months. IIRC there were other minor annoyances we managed to introduce in 0.30.0 too.

cc @wpaulino

**Comment by Kixunil** [comment]
I don't mind releasing 0.30.1 which addresses some annoyances but I wouldn't say it's super urgent.

**Reply by apoelstra** [reply]
If #1821 merges it would make me happy to have a 0.30.1.

**Reply by tcharding** [reply]
We can' technically do that can we, I thought patch bump isn't supposed to include API additions only security fixes.

**Reply by Kixunil** [reply]
We can since we're unstable.

**Reply by apoelstra** [reply]
The easiest way to think about this @tcharding is that in rust-land 0.x.y has a "major" version of x and a "minor" version of y. There is no patch version.

**Reply by wpaulino** [reply]
We would certainly appreciate a new minor release with the change, otherwise we can implement it manually for the time being as @Kixunil suggested.

**Reply by tcharding** [reply]
> The easiest way to think about this @tcharding is that in rust-land 0.x.y has a "major" version of x and a "minor" version of y. There is no patch version.

So rust-land doesn't follow the correct semantic versioning for pre 1.0? (or at least this is not how I had interpreted semantic versioning docs when I read up on it, I could have been mistaken)

**Reply by apoelstra** [reply]
@tcharding in short, correct. cargo (and by extension Rust-land) does not follow semver pre-1.0.

**Reply by stevenroose** [reply]
Interesting, I thought it was part of the semver definition that pre 1.0.0, it was actually formally `0.<major>.<minor>`. It doesn't explicitly say this, but it does say this:

> Doesn’t this discourage rapid development and fast iteration?
> 
> Major version zero is all about rapid development. If you’re changing the API every day you should either still be in version 0.y.z or on a separate development branch working on the next major version.

So that suggests that in the 0.x.y range, breaking changes in the "minor" (x) version are allowed. So it kinds implicitly confirms this approach.

(EDIT: I think that section actually is contradictory to the rest of the standard.. So it seems broken :sweat_smile: )

**Reply by Kixunil** [reply]
Indeed, it's not "incorrect", it's just additional helpful restriction.
